### PR TITLE
docs rebuild

### PR DESCRIPTION
**Pull request by szuecs**
docs: rebuild mkdocs with material, white, favicon, repo_url and edit_url
docs fix: relative paths to docs images
docs: pipeline added to create automatically PR preview and rebuild docs on merge to master
docs: add highlight.js and adjust development and install pages

preview (Zalando's only): https://skipper.docs.zalando.net/pr/843/
fix: #825 

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

### CODE REVIEW

**Comment by arjunrn** [comment]
What would be URL of the generated documentation?

**Comment by szuecs** [comment]
@arjunrn it’s not done yet, but I also hoped that someone has an idea how to do this. :)
Upload from cdp is done to s3, I don’t fully understand why, and the docs only tell about internal setup with our dev console.

**Comment by szuecs** [comment]
@arjunrn the URL mentioned in the description of the PR is working now.

**Comment by aryszka** [comment]
@szuecs @arjunrn @others what do you think having @kolja 's illustration on the main page of the docs with a humble enough size?
logo

File: docs/tutorials/basics.md [hunk 1]
```diff
@@ -91,6 +91,15 @@ Special Predicates:
 - `Path()` reduces the number of routes in O(log n) time to scan afterwards a subset in linear time
 - `PathSubtree()` reduces the number of routes O(log n) time to scan afterwards a subset in linear time
 
+#### Predicate and routing table
+
+A routing table consists of a number of routes.  A route has a list of
+predicates and filters.  Predicates match an incoming request to a
+specific, best matching, Route.  Each route has a set of filters.
```
Comment (aryszka) [inline docs/tutorials/basics.md]: "Route" -> "route" (no capital R required)

File: docs/tutorials/basics.md [hunk 2]
```diff
@@ -103,6 +112,11 @@ Some special filters are:
 - `status()` sets HTTP status code to a given value, should be used with <shunt> backend
 - `tee()` clones request to given target
 
+#### Filter in context of an HTTP request
+
+The picture shows the transformation for request and response
```
Comment (aryszka) [inline docs/tutorials/basics.md]: "the transformation of the requests and responses"

**Comment by szuecs** [comment]
I don’t mind. If you want to push something into the branch you can propose something. :)

**Comment by aryszka** [comment]
ok, i'll look at it in a bit :smile:

**Comment by aryszka** [comment]
:+1:

**Comment by aryszka** [comment]
we can also do it separately. This looks really great! Skipper is going to the 21st century!

**Comment by arjunrn** [comment]
:+1:

**Pull request merged**